app/src/main/kotlin/com/simplemobiletools/smsmessenger/helpers/ImageCompressor.kt

```diff
@@ -0,0 +1,113 @@
+package com.simplemobiletools.smsmessenger.helpers
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.BitmapFactory
+import android.graphics.Matrix
+import android.media.ExifInterface
+import android.net.Uri
+import com.simplemobiletools.commons.extensions.getCompressionFormat
+import com.simplemobiletools.commons.extensions.getMyFileUri
+import com.simplemobiletools.commons.helpers.ensureBackgroundThread
+import com.simplemobiletools.smsmessenger.extensions.extension
+import com.simplemobiletools.smsmessenger.extensions.getExtensionFromMimeType
+import com.simplemobiletools.smsmessenger.extensions.isImageMimeType
+import java.io.File
+import java.io.FileOutputStream
+
+/**
+ * Compress image to a given size based on
+ * [Compressor](https://github.com/zetbaitsu/Compressor/)
+ * */
+class ImageCompressor(private val context: Context) {
+
+    private val contentResolver = context.contentResolver
+    private val outputDirectory = File(context.cacheDir, "compressed").apply {
+        if (!exists()) {
+            mkdirs()
+        }
+    }
+
+    fun compressImage(uri: Uri, compressSize: Long, callback: (compressedFileUri: Uri?) -> Unit) {
```

Comment:
Maybe we should add an early check to see if the original file is by default smaller than the requested size? Just to avoid doing all those inputstream things and lower the chance of having some unexpected glitches.